grype/version/golang_constraint.go

```diff
@@ -31,9 +35,37 @@ func (g golangConstraint) Satisfied(version *Version) (bool, error) {
 	if g.raw == "" {
 		return true, nil // the empty constraint is always satisfied
 	}
+
+	var constraintContainsPseudoVersion bool
+	for _, units := range g.expression.units {
+		for _, unit := range units {
+			if isPseudoVersion(unit.version) {
+				constraintContainsPseudoVersion = true
+				break
+			}
+		}
+	}
+	// when we get a pseudo version from a package, and the constraint being compared against is not a pseudo version,
+	// we should not consider it as satisfied
+	// ex: constraint of type ">=v1.0.0", should not be compared to version = "v0.0.0-0.20210101000000-abcdef123456"
+	if isPseudoVersion(version.String()) && !constraintContainsPseudoVersion {
+		return false, nil
+	}
+
 	return g.expression.satisfied(version)
 }
 
+// PseudoVersionPattern is a regular expression pattern to match pseudo versions
+const PseudoVersionPattern = `^v0\.0\.0[-+].*$`
+
+// Check if a version string is a pseudo version
+func isPseudoVersion(version string) bool {
+	// List of prefixes commonly used for pseudo versions
+	regex := regexp.MustCompile(PseudoVersionPattern)
```

Comment:
This could be moved to the `PseudoVersionPattern` definition so it only is parsed once (the pattern also does not need to be exported, I think)